terraform/modules/baseline/variables.tf

```diff
@@ -525,6 +526,7 @@ variable "iam_roles" {
   description = "map of iam roles to create, where the key is the name of the role"
   type = map(object({
     assume_role_policy = list(object({
+      sid     = optional(string, null)
```

Comment:
you don't need to explicitly put null here.  Only need to specify something if you want the default to be anything other than null.